module/mempool/stdmap/incorporated_result_seals_test.go

```diff
@@ -3,12 +3,191 @@ package stdmap
 import (
 	"testing"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
+	"pgregory.net/rapid"
+
 	"github.com/onflow/flow-go/model/flow"
 	"github.com/onflow/flow-go/utils/unittest"
 )
 
+// icrSealsMachine is a description of a state machine for testing IncorporatedresultSeals
+type icrSealsMachine struct {
+	icrs  *IncorporatedResultSeals       // icrSeals being tested
+	state []*flow.IncorporatedResultSeal // model of the icrSeals
+	limit uint
+}
+
+// Init is an action for initializing a icrSeals instance.
+func (m *icrSealsMachine) Init(t *rapid.T) {
+	n := uint(rapid.IntRange(1, 1000).Draw(t, "n").(int))
+	m.icrs = NewIncorporatedResultSeals(n)
+	m.limit = n
+}
+
+// Add is a conditional action which adds an item to the icrSeals.
+func (m *icrSealsMachine) Add(t *rapid.T) {
+	i := rapid.Uint64().Draw(t, "i").(uint64)
+
+	seal := unittest.IncorporatedResultSeal.Fixture(func(s *flow.IncorporatedResultSeal) {
+		s.Header.Height = i
+	})
+
+	_, err := m.icrs.Add(seal)
+	require.NoError(t, err)
+
+	unmet := true
```

Comment:
I don't really understand why this is called `unmet`. Is this property testing terminology? A comment or more descriptive name might help.

Reply:
We're just checking the element we are adding is not already present (not previously met). Added a comment.